GDevelop.js/__tests__/GDJSSceneAsyncCodeGenerationIntegrationTests.js

```diff
@@ -0,0 +1,457 @@
+        ],
+      },
+    ]);
+
+    expect(
+      runtimeScene.getVariables().get('SuccessVariable').getAsNumber()
+    ).toBe(0);
+
+    // Process the tasks (after faking it's finished).
+    runtimeScene.getAsyncTasksManager().markAllFakeAsyncTasksAsFinished();
+    runtimeScene.getAsyncTasksManager().processTasks(runtimeScene);
+    expect(
+      runtimeScene.getVariables().get('SuccessVariable').getAsNumber()
+    ).toBe(2);
+
+    // Process the tasks (after faking it's finished).
+    runtimeScene.getAsyncTasksManager().markAllFakeAsyncTasksAsFinished();
+    runtimeScene.getAsyncTasksManager().processTasks(runtimeScene);
+    expect(
+      runtimeScene.getVariables().get('SuccessVariable').getAsNumber()
+    ).toBe(2 + 3);
+  });
+
+  it('generates async forks that share a local variable', function () {
+    scene.getVariables().insertNew('SuccessVariable', 0).setValue(0);
+    const runtimeScene = generateAndRunEventsForLayout([
+      {
+        type: 'BuiltinCommonInstructions::Standard',
+        variables: [{ name: 'MyLocalVariable', type: 'number', value: 1 }],
+        conditions: [],
+        actions: [],
+        events: [
+          {
+            type: 'BuiltinCommonInstructions::Standard',
+            conditions: [],
+            actions: [
+              {
+                type: { value: 'Wait' },
+                parameters: ['2.5'],
+              },
+              {
+                type: { value: 'SetNumberVariable' },
+                parameters: ['MyLocalVariable', '+', '1'],
+              },
+              {
+                type: { value: 'SetNumberVariable' },
+                parameters: ['SuccessVariable', '+', 'MyLocalVariable'],
+              },
+            ],
+          },
+          {
+            type: 'BuiltinCommonInstructions::Standard',
+            conditions: [],
+            actions: [
+              {
+                type: { value: 'Wait' },
+                parameters: ['1.5'],
+              },
+              // The same action is used here because the mock doesn't handle
+              // the wait duration.
+              {
+                type: { value: 'SetNumberVariable' },
+                parameters: ['MyLocalVariable', '+', '1'],
+              },
+              {
+                type: { value: 'SetNumberVariable' },
+                parameters: ['SuccessVariable', '+', 'MyLocalVariable'],
+              },
+            ],
+          },
+        ],
+      },
+    ]);
+
+    expect(
+      runtimeScene.getVariables().get('SuccessVariable').getAsNumber()
+    ).toBe(0);
+
+    // Process the tasks (after faking it's finished).
+    runtimeScene.getAsyncTasksManager().markAllFakeAsyncTasksAsFinished();
+    runtimeScene.getAsyncTasksManager().processTasks(runtimeScene);
+    expect(
+      runtimeScene.getVariables().get('SuccessVariable').getAsNumber()
+    ).toBe(2 + 3);
+  });
+
+  it('generates an async fork that shares a local variable a synced sub-event', function () {
```

Comment:
```suggestion
  it('generates an async fork that shares a local variable a non-async sub-event', function () {
```

(
synced = "synchronized", not "synchronous"
synchronous = non asynchronous
)